test/NuGet.Core.Tests/NuGet.Packaging.Test/SigningTests/RepositorySignatureInfoUtilityTests.cs

```diff
@@ -0,0 +1,302 @@
+                { "1.3.14.3.2.26", "SHA1" },
+            };
+
+            var testCertInfo = new TestRepositoryCertificateInfo()
+            {
+                ContentUrl = @"http://unit.test",
+                Fingerprints = new Fingerprints(certFingerprints),
+                Issuer = "CN=Issuer",
+                Subject = "CN=Subject",
+                NotBefore = DateTimeOffset.UtcNow,
+                NotAfter = DateTimeOffset.UtcNow
+            };
+
+            var repoCertificateInfo = new List<IRepositoryCertificateInfo>()
+            {
+                testCertInfo
+            };
+
+            var expectedAllowList = new List<CertificateHashAllowListEntry>()
+            {
+                new CertificateHashAllowListEntry(target, HashAlgorithmName.SHA256.ToString(), HashAlgorithmName.SHA256),
+                new CertificateHashAllowListEntry(target, HashAlgorithmName.SHA384.ToString(), HashAlgorithmName.SHA384),
+                new CertificateHashAllowListEntry(target, HashAlgorithmName.SHA512.ToString(), HashAlgorithmName.SHA512)
+            };
+
+            var repoSignatureInfo = new RepositorySignatureInfo(allSigned, repoCertificateInfo);
+
+            // Act
+            var settings = RepositorySignatureInfoUtility.GetSignedPackageVerifierSettings(repoSignatureInfo, commonSettings);
+
+            // Assert
+            settings.AllowUnsigned.Should().BeFalse();
+            settings.AllowNoClientCertificateList.Should().BeTrue();
+            settings.AllowNoRepositoryCertificateList.Should().BeFalse();
+            settings.RepositoryCertificateList.Should().NotBeNull();
+            settings.RepositoryCertificateList.ShouldBeEquivalentTo(expectedAllowList);
+        }
+
+        [Fact]
+        public void GetSignedPackageVerifierSettings_RepoSignatureInfoCertificateListWithMultipleEntriesCorrectlyPassedToSetting()
+        {
+            // Arrange
+            var target = VerificationTarget.Repository;
+            var commonSettings = _defaultSettings;
+            var allSigned = true;
+            var firstCertFingerprints = new Dictionary<string, string>()
+            {
+                { HashAlgorithmName.SHA256.ConvertToOidString(), $"{HashAlgorithmName.SHA256.ToString()}_first" },
+                { HashAlgorithmName.SHA384.ConvertToOidString(), $"{HashAlgorithmName.SHA384.ToString()}_first" },
+                { HashAlgorithmName.SHA512.ConvertToOidString(), $"{HashAlgorithmName.SHA512.ToString()}_first" }
+            };
+
+            var secondCertFingerprints = new Dictionary<string, string>()
+            {
+                { HashAlgorithmName.SHA256.ConvertToOidString(), $"{HashAlgorithmName.SHA256.ToString()}_second"},
+            };
+
+            var repoCertificateInfo = new List<IRepositoryCertificateInfo>()
+            {
+                new TestRepositoryCertificateInfo()
+                {
+                    ContentUrl = @"http://unit.test/1",
```

Comment:
ContentUrl must be HTTPS.  There's a [bug](https://github.com/NuGet/Home/issues/6777).